projects/components/src/datagrid/filters/datagrid-numeric-filter.component.html

```diff
@@ -1,15 +1,45 @@
 <div [formGroup]="formGroup" class="clr-row">
     <div class="clr-col-6">
-        <label class="clr-control-label"
-            >From:
-            <input class="from-input clr-input" type="number" [formControlName]="'from'" />
-        </label>
+        <vcd-number-with-unit-form-input
+            *ngIf="unit"
+            [label]="'vcd.cc.numeric.filter.from' | translate"
+            [formControlName]="'from'"
+            [unitOptions]="unitOptions"
+            [showUnlimitedOption]="false"
+            [min]="0"
+            [inputValueUnit]="unit"
+        >
+        </vcd-number-with-unit-form-input>
+        <vcd-form-input
+            *ngIf="!unit"
+            [type]="'number'"
+            [label]="'vcd.cc.numeric.filter.from' | translate"
+            [formControlName]="'from'"
+            [maxlength]="maxNumberLength"
+            (onEnter)="close()"
+            (onEscape)="close()"
+        ></vcd-form-input>
     </div>
 
     <div class="clr-col-6">
-        <label class="clr-control-label"
-            >To:
-            <input class="to-input clr-input" type="number" [formControlName]="'to'" />
-        </label>
+        <vcd-number-with-unit-form-input
+            *ngIf="unit"
+            [label]="'vcd.cc.numeric.filter.to' | translate"
+            [formControlName]="'to'"
+            [unitOptions]="unitOptions"
+            [showUnlimitedOption]="false"
+            [min]="0"
+            [inputValueUnit]="unit"
+        >
+        </vcd-number-with-unit-form-input>
+        <vcd-form-input
+            *ngIf="!unit"
+            [type]="'number'"
+            [label]="'vcd.cc.numeric.filter.to' | translate"
+            [formControlName]="'to'"
+            [maxlength]="maxNumberLength"
+            (onEnter)="close()"
```

Comment:
Why do enter and escape do the same thing? Enter should commit the change and escape should revert changes, was this a new feature you just added?

Reply:
No. The functionality to close the filter widget upon pressing both Enter and Escape keys is there in Vcd UI